### ISSUE TITLE
Backport ff31020ee651d4867c5f1910fa0ded11cf8c2b13

### ISSUE THREAD

**Issue by vchuravy** [llvm/llvm-project#54059]
/cherry-pick ff31020ee651d4867c5f1910fa0ded11cf8c2b13

Looks to be necessary to build Julia with Opaque Pointers turned on (do we backport those kinds of patches yet?)

cc: @pchintalapudi

**Comment by tstellar** [comment]
@nikic What do you think about backporting this? https://github.com/llvm/llvm-project/commit/ff31020ee651d4867c5f1910fa0ded11cf8c2b13

**Comment by nikic** [comment]
I don't think it makes sense to backport opaque pointer changes to LLVM 14. The overall state wasn't quite good enough at the time the branch was cut.
